tests/helper/kubernetes_utils.go

```diff
@@ -8,11 +8,11 @@ import (
 )
 
 // CreateRandNamespace create new project with random name in kubernetes cluster (10 letters)
-func CreateRandNamespace(context string) string {
+func CreateRandNamespace() string {
 	projectName := RandString(10)
 	fmt.Fprintf(GinkgoWriter, "Creating a new project: %s\n", projectName)
 	CmdShouldPass("kubectl", "create", "namespace", projectName)
-	CmdShouldPass("kubectl", "config", "set-context", context, "--namespace", projectName)
+	CmdShouldPass("kubectl", "config", "set-context", "--current", "--namespace", projectName)
```

Comment:
`kubectl config set-context` is not safe to use when tests are running in parallel AFAICT (either with or without ``--current`) . 

For two reasons:
- `kubectl` will prevent multiple concurrent writes which to the configuration file; it uses a `.lock` file and if the lock exists then changes to the configuration file are prevented (a non-zero error code is returned and the test will fail).
- Changing the current kube context of one automated test will affect any other tests that are running in parallel (for example, a different namespace then expected).

This use of `set-context` is failing tests for me on my local machine.

Reply:
Thanks a lot for you suggestion. I was also looking for the same kind of solution but was unable to find out from any other sources.

I will try it out for sure 👍

Reply:
@prietyc123 Please use the doc [1] along for more understanding along with @jgwest suggestion and create a separate pr that fixes this issue only.

[1] https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#the-kubeconfig-environment-variable

Reply:
Yes, It works for me. Thanks @amitkrout for making me understand more clearly over bjn.